src/jiraUpdateTicket.gs

```diff
@@ -130,7 +138,7 @@ function formatFieldValueForJira(fieldDefinition, value) {
  */
 function packageRowForUpdate(allJiraFields, headerRow, dataRow) {
   var keyFieldName = "issuekey";
-  var result = { key: null, fields: {} };
+  var result = { key: null, fields: {}, update:{} };
```

Comment:
The code to populate this "update" part of the jira api call is the fix for the bug with components and fixVersions
https://developer.atlassian.com/server/jira/platform/jira-rest-api-examples/#editing-an-issue-examples